mule-extensions-api/src/main/java/org/mule/runtime/extension/api/model/config/ImmutableConfigurationModel.java

```diff
@@ -34,6 +39,7 @@ public class ImmutableConfigurationModel extends AbstractComplexModel implements
   private final StereotypeModel stereotype;
   private final List<ParameterGroupModel> parameterGroupModels;
   private final Set<ExternalLibraryModel> externalLibraryModels;
+  private final MuleVersion minMuleVersion;
```

Comment:
same as in the other PR. Consider putting this somewhere in between `AbstractParameterizedModel` and `AbstractNamedModel`

Reply:
Do you think all parameterized models should be able to have a minMuleVersion configured? 

The ones that I chose for it are the ones that are being able to have a minMuleVersion parsed from the ModelParser interface.

Reply:
do you have an example of one we should definitively not?  The only example provided so far is parameters, but that's only about lack of use case (for now) rather than it don't making sense.

Reply:
I have checked which classes implement ParameterizedModel and the only ones that concerns me are MetadataTypeModelAdapter and MetadataTypeModelAdapterWithStereotype which are internal classes from the ast, I guess those could just always return empty() or the version "4.1.1"

Reply:
check that one with @elrodro83 . It doesn't sound like that one should be a parameterized model

Reply:
those MetadataTypeModelAdapter and MetadataTypeModelAdapterWithStereotype adapt MEtadataTypes as the name implies. What would happen if a new field is added in an infrastructure type on say, 4.6? Will it have a minMuleVersion as well? It it does, it must be made available through the adapter as well.

Reply:
Why are metadata types adapted to ParameterizedModels? I am trying to understand this in order to see how the min mule version fits here.

Reply:
@elrodro83 that doesn't make sense to me. Parameters are of a type... types themselves are not parameterizable.

Reply:
but types may have fields, and new fields may be added ad the product evolves.

Reply:
only ObjectType has fields, and those are already modeled through ObjectFieldMetadataType

Reply:
For the time being, I could just make them return empty() as their min mule version, since types do not have the notion of it, nor their fields.